modules/library/cql/src/main/java/org/geotools/filter/text/commons/ExpressionToText.java

```diff
@@ -147,15 +146,22 @@ public Object visit(Add expression, Object extraData) {
      */
     @Override
     public Object visit(Divide expression, Object extraData) {
-
         StringBuilder output = asStringBuilder(extraData);
-        expression.getExpression1().accept(this, output);
+        visitWithBrackets(expression.getExpression1(), output);
         output.append(" / ");
-        expression.getExpression2().accept(this, output);
-
+        visitWithBrackets(expression.getExpression2(), output);
         return output;
     }
 
+    private void visitWithBrackets(Expression expression, StringBuilder output) {
```

Comment:
This should have its own test, and maybe its own ticket too.

Reply:
The ParseSyntheticTest is actually already containing that test.
Would that be ok?

Reply:
I have added a separate CQL text validating the expressions structure/order.